app/src/main/java/org/eyeseetea/malariacare/data/database/datasources/ConversionLocalDataSource.java

```diff
@@ -168,7 +170,15 @@ private void convertMetaData(ConvertFromSDKVisitor converter) {
 
         int count;
         //Dataelements ordered by program.
-        List<ProgramExtended> programs = ProgramExtended.getAllPrograms();
+        List<ProgramExtended> allDhisPrograms = ProgramExtended.getAllPrograms();
```

Comment:
Instead of adding a for loop and including only those programs that match the desired condition (those who are considered as "valid", why don't just create a new DB query and get the proper list that you need?